core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -337,10 +382,14 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
     receivedTransactionalOffsetCommits = false
   }
 
-  def currentMemberMetadata: Map[String, Array[Byte]] = {
+  def currentMemberMetadata: List[JoinGroupResponseMember] = {
     if (is(Dead) || is(PreparingRebalance))
       throw new IllegalStateException("Cannot obtain member metadata for group in state %s".format(state))
-    members.map{ case (memberId, memberMetadata) => (memberId, memberMetadata.metadata(protocol.get))}.toMap
+    members.map{ case (memberId, memberMetadata) => new JoinGroupResponseMember()
+        .setMemberId(memberId)
+        .setGroupInstanceId(memberMetadata.groupInstanceId.getOrElse(JoinGroupRequest.EMPTY_GROUP_INSTANCE_ID))
```

Comment:
If this is nullable we can get rid of `EMPTY_GROUP_INSTANCE_ID` right?

Reply:
Actually, for all such `setGroupInstanceId` calls we can by default remove it since without any setters is equal to using `setGroupInstanceId(null)` right?

Reply:
Yep!